components/my-account/Lists/ViewLists/ViewLists.tsx

```diff
@@ -40,27 +42,46 @@ const ViewLists = (props: ListsProps) => {
     filter: publicRuntimeConfig.b2bList.filter,
   })
 
+  const [isLoading, setIsLoading] = useState<boolean>(false)
+
   // screen size declared
   const theme = useTheme()
   const mdScreen = useMediaQuery<boolean>(theme.breakpoints.up('md'))
   const { t } = useTranslation('common')
   // this will be used for creating new list
   const { user } = useAuthContext()
-  const userId = user?.userId
+  const userId = user?.id
 
   // fetching wishlist data
   const response = useGetWishlist(paginationState)
   const data = response.data as WishlistCollection
   const isPending = response.isPending
+  const refetch = response.refetch
 
   // edit list function
   const handleEditList = (id: Maybe<string>) => {
     console.log(id, ' edit clicked')
   }
 
   // copy list function
-  const handleCopyList = (id: Maybe<string>) => {
-    console.log(id, ' copy clicked')
+  const handleCopyList = async (id: string) => {
+    const newWishlist = data.items && data?.items.find((item: Maybe<CrWishlist>) => item?.id === id)
+    let listName = newWishlist?.name + ' - copy'
+    while (data.items && data.items.find((item: Maybe<CrWishlist>) => item?.name == listName)) {
```

Comment:
Do we need this while loop?

Reply:
if multiple copies are there, for eg.- List1, List1-copy, List1-copy-copy

Reply:
If I consider the first one as List1,
let listName = newWishlist?.name + ' - copy' makes it "List1-copy".

When we copy List1-copy, this function will again get called right ? then 
newWishlist?.name should be "List1-copy" and 

let listName = newWishlist?.name + ' - copy' makes it "List1-copy-copy".

Right ?